qutebrowser/keyinput/eventfilter.py

```diff
@@ -118,8 +120,92 @@ def eventFilter(self, obj: Optional[QObject], event: Optional[QEvent]) -> bool:
             raise
 
 
+class IMEEventHandler(QObject):
+    """Handles the cursorRectangleChanged event from the inputMethod.
+
+    This provides better input detection
+    """
+
+    _last_seen_rect: Optional[QRectF] = None
```

Comment:
You should be able to move the type annotation to where you set the variable below. Right now, this is a class variable, that is then later shadowed by an instance variable.